### PR TITLE
Module4

### PR DESCRIPTION
**Pull request by gkttk**

### CODE REVIEW

File: dao/src/main/java/com/epam/esm/constants/ApplicationConstants.java [hunk 1]
```diff
@@ -6,7 +6,13 @@
  */
 public final class ApplicationConstants {
 
+
```
Comment (NikitaUstsiushenka) [inline dao/src/main/java/com/epam/esm/constants/ApplicationConstants.java]: Please remove extra new lines

File: dao/src/main/java/com/epam/esm/dao/domain/GiftCertificateDao.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * This method get GiftCertificate entity by name.
+     *
+     * @param name GiftCertificate entity's name.
+     * @return Optional of GiftCertificate entity. If there is no GiftCertificate with given name, return Optional.empty().
+     * @since 4.0
+     */
+    Optional<GiftCertificate> findByName(String name);
+
```
Comment (NikitaUstsiushenka) [inline dao/src/main/java/com/epam/esm/dao/domain/GiftCertificateDao.java]: Please remove extra new lines

File: dao/src/main/java/com/epam/esm/dao/domain/UserDao.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * This method get User entity by login and password.
+     *
+     * @param login User entity's login.
+     * @return Optional of User entity. If there is no User with given name, return Optional.empty().
+     * @since 4.0
+     */
+  Optional<User> findByLoginAndPassword(String login, String password);
+
+
```
Comment (NikitaUstsiushenka) [inline dao/src/main/java/com/epam/esm/dao/domain/UserDao.java]: Extra new lines

File: dao/src/main/java/com/epam/esm/constants/ApplicationConstants.java (hunk 1 above)
Comment (gkttk) [reply dao/src/main/java/com/epam/esm/constants/ApplicationConstants.java]: commit 4d91c3e

File: dao/src/main/java/com/epam/esm/dao/domain/GiftCertificateDao.java (hunk 2 above)
Comment (gkttk) [reply dao/src/main/java/com/epam/esm/dao/domain/GiftCertificateDao.java]: commit 4d91c3e

**Review by NikitaUstsiushenka — APPROVED** [review]

**Pull request merged**